File: src/main/java/com/adven/concordion/extensions/exam/files/commands/CheckType.java
package com.adven.concordion.extensions.exam.files.commands;

import com.google.common.primitives.Doubles;
import com.google.common.primitives.Ints;

public enum CheckType {
    NUMBER, STRING, BOOLEAN;

    public boolean suit(String node) {
        switch (this) {
            case NUMBER:
                return isNum(node);
            case BOOLEAN:
                String val = node.toLowerCase();
                return "true".equals(val) || "false".equals(val);
            case STRING:
                return !isNum(node);
            default:
                return false;
        }
    }

    private boolean isNum(String node) {
        return Ints.tryParse(node) != null || Doubles.tryParse(node) != null;
    }
}

File: exam-core/src/main/java/com/adven/concordion/extensions/exam/core/utils/CheckType.kt
package com.adven.concordion.extensions.exam.core.utils

enum class CheckType {
    NUMBER, STRING, BOOLEAN;

    fun suit(node: String): Boolean {
        return when (this) {
            NUMBER -> isNum(node)
            BOOLEAN -> {
                val value = node.toLowerCase()
                "true" == value || "false" == value
            }
            STRING -> !isNum(node)
        }
    }

    private fun isNum(node: String): Boolean {
        return node.toIntOrNull() != null || node.toDoubleOrNull() != null
    }
}
